src/Stratis.Bitcoin.IntegrationTests/BlockStore/ReOrgRegularlySpecification_Steps.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            var thirdNodeReceivingAddress = node.FullNode.WalletManager()
+                .GetUnusedAddresses(new WalletAccountReference(WalletZero, AccountZero), 2)
+                .Skip(1).First();
+            return thirdNodeReceivingAddress;
+        }
+
+        private void third_node_mines_this_block()
+        {
+            //int feeForSecondToThirdTransaction = 4520; 
+            this.sharedSteps.MineBlocks(1, this.thirdNode, AccountZero, WalletZero, WalletPassword, this.secondNodeTransationFee.Satoshi);
+        }
+
+        private void fouth_node_confirms_it_ensures_tx_present()
+        {
+            this.sharedSteps.WaitForBlockStoreToSync(this.secondNode, this.thirdNode, this.fourthNode);
+
+            var transaction = this.fourthNode.FullNode.BlockStoreManager().BlockRepository.GetTrxAsync(this.secondNodeTransaction.GetHash()).Result;
+            transaction.Should().NotBeNull();
+            transaction.GetHash().Should().Be(this.secondNodeTransaction.GetHash());
+        }
+
+        private void selfish_node_reconnects_and_broadcasts()
+        {
+            this.selfishMiner.CreateRPCClient().AddNode(this.secondNode.Endpoint);
+            this.sharedSteps.WaitForBlockStoreToSync(this.selfishMiner, this.secondNode, this.thirdNode, this.fourthNode);
+        }
+
+        private void second_third_and_fourth_node_reorg_to_longest_chain()
+        {
+            TestHelper.WaitLoop(() => this.secondNode.FullNode.Chain.Height == this.selfishBlockHeight);
+            this.secondNode.FullNode.Chain.Height.Should().Be(this.selfishBlockHeight);
+
+            TestHelper.WaitLoop(() => this.thirdNode.FullNode.Chain.Height == this.selfishBlockHeight);
+            TestHelper.WaitLoop(() => this.fourthNode.FullNode.Chain.Height == this.selfishBlockHeight);
+        }
+
+        private void transaction_from_shorter_chain_is_missing()
+        {
+            var transaction = this.secondNode.FullNode.BlockStoreManager().BlockRepository.GetTrxAsync(this.secondNodeTransaction.GetHash()).Result;
+            transaction.Should().BeNull("longest chain comes from selfish miner and shouldn't contain the transaction made on the chain with the other 3 nodes.");
+        }
+
+        private void transaction_is_NOT_YET_returned_to_the_mem_pool()
+        {
+            this.fourthNode.CreateRPCClient().GetRawMempool()
+                .Should().NotContain(x => x == this.secondNodeTransaction.GetHash(), "it is not implemented yet.");
+        }
+
+        private void mining_continues_to_maturity_to_allow_spend()
+        {
+            var coinbaseMaturity = (int)this.secondNode.FullNode
+                .Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(coinbaseMaturity, this.secondNode, AccountZero, WalletZero, WalletPassword);
+
+            TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(this.selfishMiner));
```

Comment:
Are you sure you where not meant to call this with each node instead?

` TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.selfishMiner, this.secondNode));`